### ISSUE TITLE
Implementation of ranking loss functions 

### ISSUE THREAD

**Issue by faneshion** [faneshion/MatchZoo#298]
The ranking loss functions include rank-hinge-loss and rank-cross-entropy.

**Comment by ZizhenWang** [comment]
questions are listed below:
1. How to pass the user-defined value (like margin and neg_num) to the function? I try to define values in "losses.py" and implement a set and a get function to handle them, but it seems not safe. Another way is define a class and set these values as private,  I don't know is this way work.
2. I define a layer to slice keras tensors, to replace the Lambda layer in v1.0, I don't know is it a standard way and where to place the layer code
3. some confusing in pair-generator labels and logits shape

codes are pushed to feature/loss

**Comment by ZizhenWang** [comment]
Version 2.0 pushed, it use Lambda layer instead of SliceTensor in v1.0. Which one to choose requires more discussion.
